src/backend/benchmark/tpcc/tpcc_loader.cpp

```diff
@@ -0,0 +1,1426 @@
+int order_line_max_ol_quantity = 10;
+double order_line_min_amount = 0.01;
+size_t order_line_dist_info_length = 32;
+
+double stock_original_ratio = 0.1;
+int stock_min_quantity = 10;
+int stock_max_quantity = 100;
+int stock_dist_count = 10;
+
+double payment_min_amount = 1.0;
+double payment_max_amount = 5000.0;
+
+int stock_min_threshold = 10;
+int stock_max_threshold = 20;
+
+double new_order_remote_txns = 0.01;
+
+/////////////////////////////////////////////////////////
+// Create the tables
+/////////////////////////////////////////////////////////
+
+storage::Database* tpcc_database;
+storage::DataTable* warehouse_table;
+storage::DataTable* district_table;
+storage::DataTable* item_table;
+storage::DataTable* customer_table;
+storage::DataTable* history_table;
+storage::DataTable* stock_table;
+storage::DataTable* orders_table;
+storage::DataTable* new_order_table;
+storage::DataTable* order_line_table;
+
+const bool own_schema = true;
+const bool adapt_table = false;
+const bool is_inlined = true;
+const bool unique_index = false;
+const bool allocate = true;
+
+void CreateWarehouseTable() {
+  /*
+   CREATE TABLE WAREHOUSE (
+   W_ID SMALLINT DEFAULT '0' NOT NULL,
+   W_NAME VARCHAR(16) DEFAULT NULL,
+   W_STREET_1 VARCHAR(32) DEFAULT NULL,
+   W_STREET_2 VARCHAR(32) DEFAULT NULL,
+   W_CITY VARCHAR(32) DEFAULT NULL,
+   W_STATE VARCHAR(2) DEFAULT NULL,
+   W_ZIP VARCHAR(9) DEFAULT NULL,
+   W_TAX FLOAT DEFAULT NULL,
+   W_YTD FLOAT DEFAULT NULL,
+   CONSTRAINT W_PK_ARRAY PRIMARY KEY (W_ID)
+   );
+   */
+
+  // Create schema first
+  std::vector<catalog::Column> warehouse_columns;
+
+  auto w_id_column = catalog::Column(VALUE_TYPE_SMALLINT, GetTypeSize(VALUE_TYPE_SMALLINT), "W_ID", is_inlined);
+  warehouse_columns.push_back(w_id_column);
+  auto w_name_column = catalog::Column(VALUE_TYPE_VARCHAR, warehouse_name_length, "W_NAME", is_inlined);
+  warehouse_columns.push_back(w_name_column);
+  auto w_street_1_column = catalog::Column(VALUE_TYPE_VARCHAR, street_length, "W_STREET_1", is_inlined);
+  warehouse_columns.push_back(w_street_1_column);
+  auto w_street_2_column = catalog::Column(VALUE_TYPE_VARCHAR, street_length, "W_STREET_2", is_inlined);
+  warehouse_columns.push_back(w_street_2_column);
+  auto w_city_column = catalog::Column(VALUE_TYPE_VARCHAR, city_length, "W_CITY", is_inlined);
+  warehouse_columns.push_back(w_city_column);
+  auto w_state_column = catalog::Column(VALUE_TYPE_VARCHAR, state_length, "W_STATE", is_inlined);
+  warehouse_columns.push_back(w_state_column);
+  auto w_zip_column = catalog::Column(VALUE_TYPE_VARCHAR, zip_length, "W_ZIP", is_inlined);
+  warehouse_columns.push_back(w_zip_column);
+  auto w_tax_column = catalog::Column(VALUE_TYPE_DOUBLE, GetTypeSize(VALUE_TYPE_DOUBLE), "W_TAX", is_inlined);
+  warehouse_columns.push_back(w_tax_column);
+  auto w_ytd_column = catalog::Column(VALUE_TYPE_DOUBLE, GetTypeSize(VALUE_TYPE_DOUBLE), "W_YTD", is_inlined);
+  warehouse_columns.push_back(w_ytd_column);
```

Comment:
You can make this style of vector construction cleaner by using initializer lists:

```C++
std::vector<catalog::Column> warehouse_columns = {
  catalog::Column(...),
  catalog::Column(...),
  ...
};
```